index.js

```diff
@@ -47,51 +47,247 @@ const player = {
     { id: 1, name: 'Metal', songs: [1, 7, 4] },
     { id: 5, name: 'Israeli', songs: [4, 5] },
   ],
-  playSong(song) {
-    console.log(/* your code here */)
+  playSong({ title, album, artist, duration }) {
+    // receives a song object.
+    console.log(`Playing ${title} from ${album} by ${artist} | ${convertSecToMinFormat(duration)}.`)
   },
 }
 
+// help functions! ---------
+
+function convertSecToMinFormat(sec) {
+  // sec => mm:ss format.
+  const holeMin = Math.floor(sec / 60)
+  const secondsLeft = sec % 60
+  return formatNumber(holeMin) + ':' + formatNumber(secondsLeft)
+}
+
+function convertMinFormatToSec(minFormat) {
+  const arr = minFormat.split(':').map((str) => Number(str))
+  return 60 * arr[0] + arr[1]
+}
+
+function formatNumber(num) {
+  // to format from "6" => "06".
+  // note: only works up to 99.
+  return ('0' + num).slice(-2)
+}
+
+// ------------------
+
+function getSong(id) {
+  return getObjectFromArray(id, player.songs)
+}
+
+function getPlaylist(id) {
+  return getObjectFromArray(id, player.playlists)
+}
+
+function getObjectFromArray(objId, objectArr) {
+  // throws Error if Object does not exists.
+  const requestedObject = objectArr.find((obj) => obj.id === objId)
+  if (requestedObject === undefined) throw new Error('so such object exists')
+  return requestedObject
+}
+
+function removeSongFromSongs(id) {
+  removeObjectFromArray(id, player.songs)
+}
+
+function removeObjectFromArray(objId, objectArr) {
+  objectArr.splice(objectArr.indexOf(getObjectFromArray(objId, objectArr)), 1)
+}
+
+function removeSongFromPlaylists(id) {
+  let indexOfSong
+  for (let playlist of player.playlists) {
+    // Remember: .indexOf(element) returns -1 if the element is not present in the array.
+    if ((indexOfSong = playlist.songs.indexOf(id)) !== -1) {
+      playlist.songs.splice(indexOfSong, 1)
+    }
+  }
+}
+
+function generateNewIdInArrayOfObjects(objectArr) {
+  /* Gets the largest id and returns it + 1. 
+  i.e the returning id does not exists in the
+  array and is unique.
+  */
+  const idArray = getIdsArrayFromObjArray(objectArr)
```

Comment:
You could just call this `ids`. Variable names should describe what they store conceptually, not what their types are.